core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -392,6 +667,35 @@ protected abstract void doUpdateRoleListOfUser(String userName, String[] deleted
         return hybridRoleManager.getHybridRoleListOfUser(userName, filter);
     }
 
+    /**
+     * Only gets the internal roles of the user with internal domain name.
+     *
+     * @param userID ID of the user.
+     * @return internal roles list of the user.
+     * @throws UserStoreException An unexpected exception has occurred.
+     */
+    protected String[] doGetInternalRoleListOfUserWithID(String userID, String filter) throws UserStoreException {
+
+        if (Boolean.parseBoolean(realmConfig.getUserStoreProperty(MULIPLE_ATTRIBUTE_ENABLE))) {
```

Comment:
Since this is the userID, we don't need to consider abound Multi-attribute enable property?